sound/soc/sof/ops.h

```diff
@@ -77,7 +77,9 @@ static inline int snd_sof_dsp_core_power_up(struct snd_sof_dev *sdev,
 static inline int snd_sof_dsp_core_power_down(struct snd_sof_dev *sdev,
 					      unsigned int core_mask)
 {
-	if (sof_ops(sdev)->core_power_down)
+	/* Only power down the enabled cores */
+	core_mask = core_mask & sdev->enabled_cores_mask;
+	if (sof_ops(sdev)->core_power_down && core_mask)
```

Comment:
Hmm, doesn't this break the error path logic in topology.c:sof_core_enable() ?  If core is powered on successfully, but IPC to notify DSP fails, we will call snd_sof_dsp_core_power_down() with core powered up but the matching bit not set in enabled_cores_mask.

Reply:
Yeah, if we want to use sdev->enabled_cores_mask, we should make it reliable first. let's close it for now.